src/index.js

```diff
@@ -41,24 +41,29 @@ export default class Machine extends erx.Bus<State> {
   }
 
   goTo(next: State, data?: any): boolean {
+    const currEntries = this.transitions[this.state] || null;
+    const nextEntries = this.transitions[next] || null;
+
     if (this.transitioning) {
       return false;
     }
     const finish = (next) => {
       this.state = next;
       this.push(next);
-      const eFn = this.entries[next];
-      if (eFn != null) {
-        next = eFn(this, data);
+      if (nextEntries != null && nextEntries.before) {
```

Comment:
I think the before function should be optional, and if so, animations should run whether nextEntries.before exists or not.

Reply:
Also shouldn't it be onEnter and onExit rather than before and after?

Reply:
You're right, my initial thoughts were that you'd always have animation/onEnter/onExit in an entry object, but it makes more sense to not have a property, than to have one with an empty value. Annd yes, should fix the naming